mainnet-contracts/src/PufferProtocol.sol

```diff
@@ -347,26 +420,23 @@ contract PufferProtocol is IPufferProtocol, AccessManagedUpgradeable, UUPSUpgrad
             // nosemgrep basic-arithmetic-underflow
             $.nodeOperatorInfo[validator.node].vtBalance -= SafeCast.toUint96(vtBurnAmount);
             --$.nodeOperatorInfo[validator.node].activeValidatorCount;
+            $.nodeOperatorInfo[validator.node].numBatches -= validator.numBatches;
 
-            delete validator.node;
-            delete validator.bond;
-            delete validator.module;
-            delete validator.status;
-            delete validator.pubKey;
+            _deleteValidator(validator);
         }
 
         VALIDATOR_TICKET.burn(burnAmounts.vt);
         // Because we've calculated everything in the previous loop, we can do the burning
         PUFFER_VAULT.burn(burnAmounts.pufETH);
-        // Deduct 32 ETH from the `lockedETHAmount` on the PufferOracle
-        PUFFER_ORACLE.exitValidators(validatorInfos.length);
+        // Deduct 32 ETH per batch from the `lockedETHAmount` on the PufferOracle
+        PUFFER_ORACLE.exitValidators(validatorInfos.length, numExitedBatches);
```

Comment:
Since we are doing it in increments of 32 ETH, can we call this and provisionNode in a loop to avoid any modifications to puffer oracle, it is not an upgradeable contract, and it might not be straightforward to replace it with a new one